File: chrome-extension/icons/generate-icons.py
#!/usr/bin/env python3
"""Generate Chrome extension icons for Creative Kitchen.

Run this script after cloning the repo to create the PNG icons:
    cd chrome-extension/icons
    python3 generate-icons.py

Requires: pip install Pillow
"""

from PIL import Image, ImageDraw, ImageFont
import os

def generate_icon(size):
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)

    # Orange rounded rectangle background (#f97316)
    radius = max(2, size // 8)
    draw.rounded_rectangle([0, 0, size - 1, size - 1], radius=radius, fill='#f97316')

    # White "CK" text
    font_size = max(6, size // 3)
    try:
        font = ImageFont.truetype('/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf', font_size)
    except OSError:
        try:
            font = ImageFont.truetype('/System/Library/Fonts/Helvetica.ttc', font_size)
        except OSError:
            font = ImageFont.load_default()

    text = 'CK'
    bbox = draw.textbbox((0, 0), text, font=font)
    tw, th = bbox[2] - bbox[0], bbox[3] - bbox[1]
    x = (size - tw) // 2
    y = (size - th) // 2 - bbox[1]
    draw.text((x, y), text, fill='white', font=font)

    output = os.path.join(os.path.dirname(os.path.abspath(__file__)), f'icon{size}.png')
    img.save(output)
    print(f'  Created {output}')

if __name__ == '__main__':
    print('Generating Creative Kitchen extension icons...')
    for s in [16, 48, 128]:
        generate_icon(s)
    print('Done! Icons ready for Chrome extension.')

File: chrome-extension/icons/generate-icons.js
#!/usr/bin/env node
/**
 * Generate Chrome extension icons for Creative Kitchen.
 * 
 * Usage:
 *   cd chrome-extension/icons
 *   npm install canvas
 *   node generate-icons.js
 */

const { createCanvas } = require('canvas');
const fs = require('fs');
const path = require('path');

function generateIcon(size) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');

  // Orange rounded rectangle
  const radius = Math.max(2, Math.floor(size / 8));
  ctx.beginPath();
  ctx.moveTo(radius, 0);
  ctx.lineTo(size - radius, 0);
  ctx.quadraticCurveTo(size, 0, size, radius);
  ctx.lineTo(size, size - radius);
  ctx.quadraticCurveTo(size, size, size - radius, size);
  ctx.lineTo(radius, size);
  ctx.quadraticCurveTo(0, size, 0, size - radius);
  ctx.lineTo(0, radius);
  ctx.quadraticCurveTo(0, 0, radius, 0);
  ctx.closePath();
  ctx.fillStyle = '#f97316';
  ctx.fill();

  // White "CK" text
  const fontSize = Math.max(8, Math.floor(size / 2.8));
  ctx.font = `bold ${fontSize}px sans-serif`;
  ctx.fillStyle = '#ffffff';
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';
  ctx.fillText('CK', size / 2, size / 2 + 1);

  const outPath = path.join(__dirname, `icon${size}.png`);
  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync(outPath, buffer);
  console.log(`  Created ${outPath}`);
}

console.log('Generating Creative Kitchen extension icons...');
[16, 48, 128].forEach(generateIcon);
console.log('Done!');
